Framework/API/inc/MantidAPI/IFunction_fwd.h

```diff
@@ -0,0 +1,45 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2015 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef IFUNCTION_FWD_H_
+#define IFUNCTION_FWD_H_
+
+#include <boost/shared_ptr.hpp>
+#include <memory>
+
+namespace Mantid {
+namespace API {
+
+/**
+  This file provides forward declarations for Mantid::API::IFunction
+*/
+
+/// forward declare of Mantid::API::IFunction
+class IFunction;
+/// Shared pointer to Mantid::API::IFunction
+using IFunction_sptr = boost::shared_ptr<IFunction>;
```

Comment:
Can we get away with making these pointers unique rather than shared? If we need the shared ownership then obviously not but just thought I would check.

Reply:
I am not sure that it will be easy. At least not in this PR.